svg_test.go

```diff
@@ -137,3 +140,31 @@ func (s *SVGSuite) TestProcessIcon(c *gc.C) {
 		}
 	}
 }
+
+func (s *SVGSuite) TestSetXMLAttr(c *gc.C) {
+	expected := []xml.Attr{
+		{
+			Name: xml.Name{
+				Local: "id",
+			},
+			Value: "foo",
+		},
+	}
+
+	result := setXMLAttr([]xml.Attr{}, xml.Name{
+		Local: "id",
+	}, "foo")
+	c.Assert(result, gc.DeepEquals, expected)
+
+	result = setXMLAttr([]xml.Attr{
```

Comment:
Perhaps add another attr here to check that the extraneous attributes are not removed in the process?